### DISCUSSION TITLE
secure way to pass jwt token in header?

### DISCUSSION THREAD

**Discussion by GorvGoyl** [supabase/supabase#10865] [Questions] [answered]
Access token retrieved from `supabase.auth.session().access_token` can be directly decoded without the need of any extra keys. I tested it on https://jwt.io/. Is this what we suppose to send in the API header `Authorization: Bearer ${accessToken}` without further encoding it?

**Comment by GaryAustin1** [comment, marked as answer]
jwt's are not hidden, they just are encoded so you can be sure they have not been modified.
Https headers are encoded for transmission just like the body.

**Reply by GorvGoyl** [reply]
I see, is there a way to make it more secure?  the CTO of supabase In one of his videos, mentioned signing jwt token with public key that can only be decoded by a private key on the server side. Is this something supabase support natively?

**Reply by GaryAustin1** [reply]
I'm not sure of the context.  But the jwt is signed by a private key that can only be officially verified if you know that private key.  The jwts issued by Supabase can't be modified and are automatically verified to be the same as the one issued anytime you use them in the Supabase platform.

If you notice on jtw.io usually you see a "invalid signature" when you just enter a token.  You can get the signature from your instance settings and enter it (if you trust jwt.io) to validate the jwt.


 https://jwt.io/introduction
A pretty good description.  Supabase auth uses signed tokens.

**Reply by GorvGoyl** [reply]
Thanks, I missed the part about "invalid signature". so the jwt token can be decoded by anyone but its integrity can only be verified by private key which was used to sign the token in the first place.

**Reply by GaryAustin1** [reply]
Yes.  If they get the jwt they can decode it.  It is not usually available in transit (https headers are encrypted), but is still "safe" if exposed.
